### PR TITLE
Reactors from Charleston Instance

### PR DESCRIPTION
**Pull request by BrOgBr**
**These Reactors are from Charleston Instance.**

* **Server Mode**:  re

* **Description of Pull Request**:  Added items (reactors) from Charleston instance.

https://www.novaragnarok.com/forum/news/excellion-patch-notes-r163/

I've used the DivinePride and NovaRO for do it.

### CODE REVIEW

**Comment by anacondaqq** [comment]
Are you sure that these ID's range are correct? I don't think so, but can be wrong.

**Comment by BrOgBr** [comment]
I've used the DivinePride's DB and NovaRO to collect the infos but I'm not sure, all are working perfectly here. :)

**Comment by anacondaqq** [comment]
@BrOgBr I have manually re-check the item range at the official server, and it is correct in terms of "official source". But not correct from the logical side from "officials", because this range has been reserved for another thing long time ago, not for these stones. That why I asked about the range and re-check it. Anyway, I'm sure that this is official bug and error, but nothing can say about PR because it follows officials and it is correct. Very confusing thing. Maybe "officials" changed something not long time ago, dunno... 😞

File: db/re/item_db.txt [hunk 1]
```diff
@@ -3119,6 +3119,30 @@
...
+4976,Earth_Resistance_Reactor,Earth Resistance Reactor,6,0,,,,,,,,,,,,,,,{ bonus2 bSubEle,Ele_Earth,25; bonus2 bSubEle,Ele_Fire,-25; },{},{}
+4977,Wind_Resistance_Reactor,Wind Resistance Reactor,6,0,,,,,,,,,,,,,,,{ bonus2 bSubEle,Ele_Wind,25; bonus2 bSubEle,Ele_Earth,-25; },{},{}
+4978,Recovery_Reactor_101,Recovery Reactor 101,6,0,,,,,,,,,,,,,,,{ if (getrefine()>=7) bonus2 bHPRegenRate,100,5000; else bonus2 bHPRegenRate,50,5000; },{},{}
+4979,Recovery_Reactor_102,Recovery Reactor 102,6,0,,,,,,,,,,,,,,,{ if (getrefine()>=7) bonus2 bSPRegenRate,5,5000; else bonus2 bHPRegenRate,3,5000; },{},{}
+4980,Recovery_Reactor_201,Recovery Reactor 201,6,0,,,,,,,,,,,,,,,{ if (getrefine()>=7) bonus2 bRegenPercentHP,100,5000; else bonus2 bRegenPercentHP,50,5000; },{},{}
+4981,Recovery_Reactor_202,Recovery Reactor 202,6,0,,,,,,,,,,,,,,,{ if (getrefine()>=7) bonus2 bRegenPercentSP,100,5000; else bonus2 bRegenPercentSP,50,5000; },{},{}
+4982,STR_Supplement_Reactor,STR Supplement Reactor,6,0,,,,,,,,,,,,,,,{ if (getrefine()>=7) bonus bBaseAtk,10*(readparam(bStr)/10); else bonus bBaseAtk,5*(readparam(bStr)/10); },{},{}
```
Comment (Atemo) [inline db/re/item_db.txt]: https://www.divine-pride.net/database/item/4982
If the refine is higher than 7 the bonus give additional ATK+10. It should be
`if (getrefine()>=7) { .@bonus = 10 } bonus bBaseAtk,.@bonus + 5*(readparam(bStr)/10);`

File: db/re/item_db.txt [hunk 2]
```diff
@@ -3119,6 +3119,30 @@
...
+4983,INT_Supplement_Reactor,INT Supplement Reactor,6,0,,,,,,,,,,,,,,,{ if (getrefine()>=7) bonus bMatk,10*(readparam(bInt)/10); else bonus bMatk,5*(readparam(bInt)/10); },{},{}
```
Comment (Atemo) [inline db/re/item_db.txt]: https://www.divine-pride.net/database/item/4983
Same thing as above, additional MATK+10 when refine +7 or higher.

File: db/re/item_db.txt [hunk 3]
```diff
@@ -3119,6 +3119,30 @@
...
+4984,DEF_Supplement_Reactor,DEF SupplementReactor,6,0,,,,,,,,,,,,,,,{ bonus bDef,100; },{},{}
+4985,PD_Supplement_Reactor,PD Supplement Reactor,6,0,,,,,,,,,,,,,,,{ bonus bFlee2,3; },{},{}
+4986,Attack_Supplement_Reactor,Attack Supplement Reactor,6,0,,,,,,,,,,,,,,,{ bonus bBaseAtk,20; },{},{}
+4987,Magic_Supplement_Reactor,Magic Supplement Reactor,6,0,,,,,,,,,,,,,,,{ bonus bMatk,20; },{},{}
+4988,HP_Supplement_Reactor,HP Supplement Reactor,6,0,,,,,,,,,,,,,,,{ bonus bMaxHPrate,5; },{},{}
+4989,SP_Supplement_Reactor,SP Supplement Reactor,6,0,,,,,,,,,,,,,,,{ bonus bMaxSPrate,3; },{},{}
+4990,Frozen_Supplement_Reactor,Frozen Supplement Reactor,6,0,,,,,,,,,,,,,,,{ bonus2 bResEff,Eff_Freeze,10000; },{},{}
+4991,ASPD_Supplement_Reactor,ASPD Supplement Reactor,6,0,,,,,,,,,,,,,,,{ bonus bAspd,1; },{},{}
+4992,HPAbsorb1_Supplement_Reactor,HP Absorb Supplement Reactor,6,0,,,,,,,,,,,,,,,{ bonus2 bHPDrainRate,1,1; },{},{}
+4993,SPAbsorb1_Supplement_Reactor,SP Absorb Supplement Reactor,6,0,,,,,,,,,,,,,,,{ bonus2 bSPDrainRate,1,1; },{},{}
```
Comment (Atemo) [inline db/re/item_db.txt]: https://www.divine-pride.net/database/item/4992
https://www.divine-pride.net/database/item/4993
```
bonus2 bHPDrainRate,x,n;		Adds a x/10% chance to drain n% HP from inflicted damage when attacking
bonus2 bSPDrainRate,x,n;		Adds a x/10% chance to drain n% SP from inflicted damage when attacking
```
1% chance to get 1%, it should be :
`bonus2 bHPDrainRate,10,1;`
`bonus2 bSPDrainRate,10,1;`

File: db/re/item_db.txt (hunk 1 above)
Comment (BrOgBr) [reply db/re/item_db.txt]: I'll do it now, thank you @Atemo

File: db/re/item_db.txt (hunk 3 above)
Comment (BrOgBr) [reply db/re/item_db.txt]: Haha thanks!
My bad. :B

**Pull request merged**